src/Dialect/ONNX/Transforms/ConstProp.cpp

```diff
@@ -134,6 +145,46 @@ Value ConstZeroTensor(
           type, rewriter.getZeroAttr(type.getElementType())));
 }
 
+template <typename GetFPConstFunc =
+              std::function<APFloat(const llvm::fltSemantics &, bool)>,
+    typename GetIntConstFunc = std::function<APInt(unsigned)>>
+Value GetClipConstantOfType(PatternRewriter &rewriter, ShapedType type,
+    Value value, GetFPConstFunc fpConstantFunc, bool isNegative,
+    GetIntConstFunc intConstantFunc) {
+  auto elemType = type.getElementType();
+  if (auto floatType = dyn_cast<FloatType>(elemType)) {
+    auto fpValue =
+        fpConstantFunc(floatType.getFloatSemantics(), /*Negative=*/isNegative);
+    return rewriter.create<ONNXConstantOp>(value.getLoc(), Attribute(),
+        DenseElementsAttr::get(type, llvm::ArrayRef(fpValue)));
+  }
+  auto intValue = intConstantFunc(elemType.getIntOrFloatBitWidth());
+  return rewriter.create<ONNXConstantOp>(value.getLoc(), Attribute(),
```

Comment:
Just check: I don't see the `isNegative` being used on the int path. It looks ok as the `isNegative` is only used to find the minimum floating point.

Reply:
Exactly. APInt has different methods for Min and Max, while APFloat has only `getLargest`, which makes sense since the latter only changes one bit, while the former have different ranges for positive and negative.